### ISSUE TITLE
How can I find all reference types within a lambda

### ISSUE THREAD

**Issue by gitlui** [javaparser/javaparser#2234]
So I am currently trying to build a tool to check if specific dependency rules are violated. Therefore iam gathering all dependencies of a class to other classes. 
This worked so far but now I am stuck with how to get all the referenced Types in a lambda.

For Example the following:
`Consumer<String> consumer = (name) -> new DDDModelClass(name).doSomething();`
or like a static method call whose returntype is a ReferenceType:
`Consumer<String> consumer1 = (name) -> staticDependencyMethod(name).doSomething();
`
I tried to visit the lambda body recursively with the visitor which found the lambda in the first place like so:
```
@Override
public void visit(LambdaExpr lambdaExpr, List<Dependency> dependencies) {
    super.visit(lambdaExpr, dependencies);
    lambdaExpr.getBody().accept(this, dependencies);
}
```
but that didnt work

**Comment by gitlui** [comment]
Ok I just found out, that I can already find the Type of the 
`Consumer<String> consumer1 = (name) -> staticDependencyMethod(name).doSomething();`
but the constructor call is still a problem

**Comment by matozoid** [comment]
Hi!
Here's how you find out which part of the source code ends up where in the AST: https://javaparser.org/inspecting-an-ast/
It's always recommended to read the book too, it will explain the symbol solver, which I think you will want to use after a while.

**Comment by gitlui** [comment]
Thank you very much. The printing of the AST as yml helped a lot :) I just needed to override the visitor for ClassOrInterfaceType and I got the information I needed.
I already used the symbol solver for some of the other dependencies and it works great so far and both the java parser and the symbol solver are really easy to use. 
Keep up the good work and thank you for open sourcing!
I might even start to contribute later on, because it really is an interesting topic.
